### PR TITLE
feat: add rds integration, logging and validation

### PR DESCRIPTION
**Pull request by Arestout**
Frontend:

https://d3qsyl0aw7gjne.cloudfront.net/
POST for adding products: https://d3qsyl0aw7gjne.cloudfront.net/admin/product-form/
PR: https://github.com/Arestout/node-aws-shop/pull/3

Endpoints:
  GET - https://pjberqbx41.execute-api.eu-west-1.amazonaws.com/dev/products
  GET - https://pjberqbx41.execute-api.eu-west-1.amazonaws.com/dev/products/{productId}
  POST - https://pjberqbx41.execute-api.eu-west-1.amazonaws.com/dev/products

Swagger:  https://app.swaggerhub.com/apis/Arestout/book-shop/1.0.0

+ SQL initial scripts
+ RDS integration 
+ POST/products body validation
+ error handling
+ logging incomming requests
+ transaction based creation of product

### CODE REVIEW

**Review by generalov — APPROVED** [review]
LGTM

**Comment by generalov** [comment]
Why does Swagger only render one object returned from the `/products` endpoint?
image

**Comment by generalov** [comment]
I cannot create a new product using frontend due to the CORS error.
image

**Comment by Arestout** [comment]
> 
> 
> I cannot create a new product using frontend due to the CORS error.
> image

should be fixed now

File: product-service/handlers/postProduct.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    await client.query('COMMIT');
+
+    const { rows: product } = await client.query(`
+    SELECT p.*, s.count FROM products p LEFT JOIN stocks s ON p.id = s.product_id WHERE p.id = $1`,
+    [productId]);
+  
+    
+    if (!product) {
+      return {
+        statusCode: 403,
```
Comment (candentira) [inline product-service/handlers/postProduct.ts:39]: Please return 400 status code. 403 is used when user is not authorized to perform this action.

File: product-service/handlers/postProduct.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        headers: {
+          'Access-Control-Allow-Origin': '*',
+          'Access-Control-Allow-Credentials': true,
+        },
+        body: JSON.stringify(`Product creation failed`)
+      }
+    }
+    
+    return {
+      statusCode: 200,
```
Comment (candentira) [inline product-service/handlers/postProduct.ts:49]: Please return 201 status code upon successful creation of the resource.

**Pull request merged**